pallets/accounts/src/lib.rs

```diff
@@ -573,22 +589,46 @@ pub mod pallet {
 				}
 			}
 
-			// The lines below only execute if the number of votes received by `data_hash` passes the `threshold`)
+			// The lines below only execute if the number of votes received by `data_hash` passes the `threshold`
 
 			let current_block_number = <frame_system::Pallet<T>>::block_number();
 
 			let lock_period: U256 =
 				data.lock_period.try_into().map_err(|_| Error::<T>::SystematicFailure)?;
 			let frag_amount: <T as pallet_assets::Config>::Balance = data_amount.saturated_into();
 
-			// TODO - get FRAG price from oracle
-			/* NOTE: this is assigning 20% of Tickets and NOVA.
-			 */
-			let percentage_amount = Self::apply_20_percent(data_amount);
+			let current_frag_price = Self::get_oracle_price();
+			// Calculate the initial amount of Tickets and NOVA to convert based on
+			// 1) the amount of FRAG locked, 2) the initial percentages defined in Config, 3) the current FRAG price
+			let initial_nova_amount = Self::initial_amount(
+				data_amount,
+				T::InitialPercentageNova::get(),
+				current_frag_price,
+			);
+			let initial_tickets_amount = Self::initial_amount(
+				data_amount,
+				T::InitialPercentageTickets::get(),
+				current_frag_price,
+			);
+			// convert
+			let nova_amount: <T as pallet_balances::Config>::Balance =
+				initial_nova_amount.saturated_into();
+			let tickets_amount: <T as pallet_assets::Config>::Balance =
+				initial_tickets_amount.saturated_into();
 
-			// TODO - apply mechanism of periodic vesting of NOVA and Tickets
-			let nova_amount: <T as pallet_balances::Config>::Balance = percentage_amount.saturated_into();
-			let tickets_amount: <T as pallet_assets::Config>::Balance = percentage_amount.saturated_into();
+			// need to track when we received the first Lock event from the user.
+			// This is needed in the case the user locks some FRAG, unlock and then Lock again later.
+			let mut first_lock_block_num = current_block_number;
```

Comment:
not sure about this, how can they do that? locks last a definite amount of time

Reply:
I am thinking about the case of users who lock some FRAG, unlock once the lock period is over and then lock other FRAG again, but never actually link their account on Clamor. 
Probably overthinking....let me delete this.

Reply:
ah @sinkingsugar I forgot also to mention that I introduced this `first_lock_block_num` because we use the same structure `EthLockedFrag` for both Lock and Unlock events. So, when the user decide to withdraw, without this field, we would not be able to know the block number when the lock of FRAG happened, and so not able to calculate how much withdraw.

Reply:
good point!